codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/RegisterServiceMetadataMiddleware.java

```diff
@@ -0,0 +1,110 @@
+package software.amazon.smithy.aws.go.codegen;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import software.amazon.smithy.aws.traits.ServiceTrait;
+import software.amazon.smithy.aws.traits.auth.SigV4Trait;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.GoDelegator;
+import software.amazon.smithy.go.codegen.GoSettings;
+import software.amazon.smithy.go.codegen.SymbolUtils;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.go.codegen.integration.MiddlewareRegistrar;
+import software.amazon.smithy.go.codegen.integration.RuntimeClientPlugin;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.utils.ListUtils;
+
+public final class RegisterServiceMetadataMiddleware implements GoIntegration {
+    List<RuntimeClientPlugin> runtimeClientPlugins = new ArrayList<>();
+
+    @Override
+    public byte getOrder(){
+        return 30;
+    }
```

Comment:
`30` is same order as AssembleMiddlewareStack, will that cause any ordering issues?

Reply:
No, previous PR's assigned AssembleMiddlewareStack -40 as order ID.